### ISSUE TITLE
E203 fix should leave equal amounts of space on either side of a slice operator

### ISSUE THREAD

**Issue by bersbersbers** [astral-sh/ruff#7259]
```python
ham[lower + offset : upper + offset]
```

`ruff --fix --select E203 bug.py` (v0.0.287) gives

```python
ham[lower + offset: upper + offset]
```

which is discouraged by PEP 8:

> However, in a slice the colon acts like a binary operator, and should have equal amounts on either side (treating it as the operator with the lowest priority).

https://peps.python.org/pep-0008/#pet-peeves

So, `ham[lower + offset: upper + offset]` is incorrect. I would argue that `ham[lower + offset : upper + offset]` (which is what `black` outputs) should not be flagged by `E203`; and if so, *both* spaces should be removed (but I still argue to remove none at all).

**Comment by charliermarsh** [comment]
Our behavior is consistent with pycodestyle, but I am inclined to support this (though it's low priority).

**Comment by oefe** [comment]
It's also inconsistent with black, which puts spaces on both sides.
(Though I would argue that if you're using black, you don't that rule anyway.)

**Comment by gertvdijk** [comment]
I think this is a duplicate of https://github.com/astral-sh/ruff/issues/6861.

**Comment by charliermarsh** [comment]
Gonna look at fixing this today.
